include/sqlpp11/result.h

```diff
@@ -121,29 +133,35 @@ namespace sqlpp
 
       iterator& operator++()
       {
-        _result.get().next(_result_row.get());
+        _result_ptr->next(*_result_row_ptr);
```

Comment:
Note: This is undefined behavior for default-constructed iterators.

Reply:
Assuming that `result_t::iterator` is modelling [LegacyInputIterator](https://en.cppreference.com/w/cpp/named_req/InputIterator), then that page says that for the expression `++r`, there is a precondition that the iterator `r` must be dereferenceable. Obviously default-constructed iterators are not dereferenceable, so calling `operator++` for a default-constructed iterator is undefined behavior per se. With the current code the most likely result will be segfault when the program tries to derefence the NULL pointer.

I think that it makes sense to leave it this way because it signals a bug in the caller's code. Alternatively we can either throw an exception, or just ignore the increment, the last approach IMHO being the worst one. What is your suggestion?

Reply:
I am fine leaving it this way, just wanted to make sure this is not accidental.